tehuti-vis.py

```diff
@@ -17,78 +17,95 @@
 # along with Tehuti.  If not, see <http://www.gnu.org/licenses/>.
 import argparse
 
-import matplotlib.pyplot as plt
+from tehuti import Results
+from vis_methods import VaryRepoCommit, Violin
 
-from tehuti import Results, sha, shorten_sha
 
+class Vis(object):
+    """Provides functionality to visualise tehuti metrics results."""
 
-# Based on (minor label tweaks and line plotting on single value input):
-# http://pyinsci.blogspot.co.uk/2009/09/violin-plot-with-matplotlib.html
-def violin_plot(ax,data,bp=False):
-    '''
-    create violin plots on an axis
-    '''
-    from scipy.stats import gaussian_kde
-    from numpy import arange
-    pos = range(len(data))
-    dist = max(pos)-min(pos)
-    w = min(0.15*max(dist,1.0),0.5)
-    for d,p in zip(data,pos):
-        # If we only have a single value, plot a line instead
+    def __init__(self, results, method):
+        """
+        Construct a tehuti metrics visualiser.
+
+        The data selected and plots produced are determined by the
+        visualisation method (or state) that the visualiser is constructed
+        with.
+
+        Args:
+
+        * results:
+            The benchmarking results to visualise.
+        * method:
+            The visualisation method (state) to use. Available states are:
+                * 'basic': plot benchmark results against repository commits
+                           for all supplied metrics.
+                * 'violin': plot benchmark results as a violin plot.
+        """
+        self.results = results
+        self._method = method
+        self.plot_data = None
+
+        self.methods = {'basic': VaryRepoCommit(self),
+                        'violin': Violin(self)}
+
+    @property
+    def method(self):
+        if self._method is None:
```

Comment:
@dkillick How can this be `None`? I'm guessing only if the instance has been instantiated explicitly by the user with a call like `Vis(results, None)` ... seems confusing. Does this mean that `method` in the `__init__` is a kwarg? I'm thinking not ... 

How about removing this check in here and putting it in the `__init__` ?